src/openzaak/utils/inclusion.py

```diff
@@ -0,0 +1,222 @@
+            ):
+                yield entry
+
+    def _some_related_field_inclusions(
+        self, path, field, instance, inclusion_serializer
+    ):
+        if self.allowed_paths is not None and path not in self.allowed_paths:
+            return []
+
+        # Properly include loose fk fields
+        if (
+            isinstance(field, FKOrURLField)
+            or isinstance(field, LooseFKHyperlinkedRelatedField)
+            or isinstance(field, LooseFKHyperlinkedIdentityField)
+        ):
+            return self._loose_fk_field_inclusions(
+                path, field, instance, inclusion_serializer
+            )
+        elif isinstance(field, HyperlinkedRelatedField):
+            return self._primary_key_related_field_inclusions(
+                path, field, instance, inclusion_serializer
+            )
+        return super()._some_related_field_inclusions(
+            path, field, instance, inclusion_serializer
+        )
+
+    def _loose_fk_field_inclusions(self, path, field, instance, inclusion_serializer):
+        value = field.get_attribute(instance)
+        # In case it's an external resource
+        if isinstance(value, str) or hasattr(value, "_initial_data"):
+            if self._has_been_seen_external(value):
+                return
+
+            try:
+                yield getattr(instance, field.field_name)  # ._initial_data
+            except FetchError:  # Something failed during fetching, ignore this instance
+                return []
+        else:
+            for entry in self._primary_key_related_field_inclusions(
+                path, field, instance, inclusion_serializer
+            ):
+                yield entry
+
+
+class InclusionJSONRenderer(_InclusionJSONRenderer, CamelCaseJSONRenderer):
+    """
+    Ensure that the InclusionJSONRenderer produces camelCase and properly loads loose fk
+    objects
+    """
+
+    loader_class = InclusionLoader
+    response_results_key = "results"
+    response_data_key = "data"
+
+    @use_connection_pool
+    def render(self, data, accepted_media_type=None, renderer_context=None):
+        # Only render inclusions for list operation
+        if renderer_context["view"].action == "list":
+            return super().render(data, accepted_media_type, renderer_context)
+        return super(_InclusionJSONRenderer, self).render(
+            data, accepted_media_type, renderer_context
+        )
+
+    def _render_inclusions(self, data, renderer_context):
+        render_data = super()._render_inclusions(data, renderer_context)
+
+        # Store serializer data in `results` for paginated lists, instead of `data`
+        # to prevent breaking changes
+        if (
+            renderer_context["view"].action == "list"
+            and render_data
+            and "count" in render_data
+        ):
+            render_data[self.response_results_key] = render_data.pop(
+                self.response_data_key
+            )
```

Comment:
I'm not sure what is causing this to happen? Can you point me to the relevant code?

Reply:
This is no longer relevant, I made the key of the data/inclusions configurable in the library